src/coreclr/vm/comutilnative.cpp

```diff
@@ -1106,21 +1104,14 @@ extern "C" void QCALLTYPE GCInterface_UnregisterFrozenSegment(void* segment)
 **Arguments: Object of interest
 **Exceptions: None
 ==============================================================================*/
-FCIMPL1(void, GCInterface::SuppressFinalize, Object *obj)
+extern "C" void QCALLTYPE GCInterface_SuppressFinalize(QCall::ObjectHandleOnStack pObj)
 {
-    FCALL_CONTRACT;
+    QCALL_CONTRACT_NO_GC_TRANSITION;
 
     // Checked by the caller
-    _ASSERTE(obj != NULL);
-
-    if (!obj->GetMethodTable ()->HasFinalizer())
-        return;
-
-    GCHeapUtilities::GetGCHeap()->SetFinalizationRun(obj);
-    FC_GC_POLL();
```

Comment:
I think it would be fine to keep this as is and just delete `FC_GC_POLL`.

The problem with missing polls that can lead to (often theoretical) GC starvation is universal across all FCalls. If we wanted to fix it, we would want to teach the JIT to emit the poll. It does not make to make one-off fixes like this one.

Reply:
> If we wanted to fix it, we would want to teach the JIT to emit the poll.

`GTF_CALL_M_NOGCCHECK` flag that the JIT sets on FCalls takes care of it for the most part, but it is not perfect.

Reply:
> just delete FC_GC_POLL.

Also, do the same for `FC_GC_POLL_RET`s in comutilnative.cpp.